src/NServiceBus.Core/obsoletes.cs

```diff
@@ -347,6 +347,14 @@ public static void RegisterEncryptionService(this EndpointConfiguration config,
             throw new NotImplementedException();
         }
     }
+
+    [ObsoleteEx(
+        Message = "If self-hosting, call your start/stop code before/after `Endpoint.Create` or `Endpoint.Stop`. If using NServiceBus.Host or NServiceBus.Host.AzureCloudService, implement `IWantToRunWhenEndpointStartsAndStops` instead",
```

Comment:
Perhaps something like:

IWantToRunWhenBusStartsAndStops has been removed from the core and moved to the host implementations. If you're self-hosting, instead of using this interface, you can call any startup code right beforeEndpoint.Create or any cleanup code right afterEndpoint.Stop. When using either NServiceBus.Host or NServiceBus.Host.AzureCloudService, use the host's interface definition of the `IWantToRunWhenBusStartsAndStops` instead.